org.opent2t.sample.binaryswitch.superpopular/com.contosothings.binaryswitch/js/thingTranslator.js

```diff
@@ -1,5 +1,14 @@
 'use strict';
 var OpenT2TError = require('opent2t').OpenT2TError;
+var crypto = require('crypto');
+
+/**
+ * Generate a GUID for a given ID.
+ */
+function generateGUID(stringID) {
+    var guid = crypto.createHash('sha1').update('Contoso' + stringID).digest('hex');
+    return guid.substr(0, 8) + '-' + guid.substr(8, 4) + '-' + guid.substr(12, 4) + '-' + guid.substr(16, 4) + '-' + guid.substr(20, 12);
```

Comment:
Got it! Fixed.

---
In reply to: [108541340](https://github.com/openT2T/translators/pull/217#discussion_r108541340) [](ancestors = 108541340,108326264,108277675)